pkg/clusteragent/custommetrics/provider_test.go

```diff
@@ -0,0 +1,165 @@
+	namespace string
+	labels    labels.Set
+}
+
+func TestGetExternalMetric(t *testing.T) {
+
+	metricName := "m1"
+	goodLabel := map[string]string{
+		"foo": "bar",
+	}
+	badLabel := map[string]string{
+		"oof": "bar",
+	}
+	ns := "default"
+	tests := []struct {
+		name          string
+		metricsStored []ExternalMetricValue
+		compared      metricCompare
+		expected      []external_metrics.ExternalMetricValue
+	}{
+		{
+			"nothing stored",
+			nil,
+			metricCompare{},
+			[]external_metrics.ExternalMetricValue{},
+		},
+		{
+			"one matching metric stored",
+			[]ExternalMetricValue{
+				{
+					MetricName: metricName,
+					Labels:     goodLabel,
+					HPA:        ObjectReference{Namespace: ns},
+					Valid:      true,
+				}},
+			metricCompare{
+				provider.ExternalMetricInfo{Metric: metricName},
+				ns,
+				goodLabel,
+			},
+			[]external_metrics.ExternalMetricValue{
+				{
+					MetricName:   metricName,
+					MetricLabels: goodLabel,
+				}},
+		},
+		{
+			"one non matching metric stored",
+			[]ExternalMetricValue{
+				{
+					MetricName: metricName,
+					Labels:     goodLabel,
+					HPA:        ObjectReference{Namespace: ns},
+					Valid:      true,
+				}},
+			metricCompare{
+				provider.ExternalMetricInfo{Metric: metricName},
+				ns,
+				badLabel,
+			},
+			[]external_metrics.ExternalMetricValue{},
+		},
+		{
+			"one matching name metrics stored",
+			[]ExternalMetricValue{
+				{
+					MetricName: metricName,
+					Labels:     goodLabel,
+					HPA:        ObjectReference{Namespace: ns},
+					Valid:      true,
+				}, {
+					MetricName: metricName,
+					Labels:     badLabel,
+					HPA:        ObjectReference{Namespace: ns},
+					Valid:      true,
+				}},
+			metricCompare{
+				provider.ExternalMetricInfo{Metric: metricName},
+				ns,
+				badLabel,
+			},
+			[]external_metrics.ExternalMetricValue{
+				{
+					MetricName:   metricName,
+					MetricLabels: goodLabel,
+				},
+			},
+		},
+		{
+			"one non matching labels metrics stored",
+			[]ExternalMetricValue{
+				{
+					MetricName: metricName,
+					Labels:     goodLabel,
+					HPA:        ObjectReference{Namespace: ns},
+					Valid:      true,
+				}, {
+					MetricName: metricName,
+					Labels:     goodLabel,
+					HPA:        ObjectReference{Namespace: ns},
+					Valid:      true,
+				}},
+			metricCompare{
+				provider.ExternalMetricInfo{Metric: metricName},
+				ns,
+				badLabel,
+			},
+			[]external_metrics.ExternalMetricValue{},
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			m := &mockDatadogProvider{
+				mockListAllExternalMetricValues: func() ([]ExternalMetricValue, error) {
+					return test.metricsStored, nil
+				},
+			}
+			dp := datadogProvider{
+				store: Store(m),
+			}
+			output, err := dp.GetExternalMetric(test.compared.namespace, test.compared.labels.AsSelector(), test.compared.name)
+			require.NoError(t, err)
+			require.Equal(t, len(output.Items), len(test.expected))
```

Comment:
nit: the argument order for testify asserts/requires is `t,expected,actual` : https://godoc.org/github.com/stretchr/testify/require#Equal

Reply:
ah, right, good catch

Reply:
Missed a few others :)